From c4531d607053c0c7b61260e3e6f5d6394479f1e4 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 12 May 2016 11:52:14 -0700
Subject: [PATCH 0426/1239] Get lmac_type from LMAC0/LMAC2 based on the DLM
 used.

BDK has been updated to save lmac_type in BGXX_CMR{0,2}_CONFIG in CSR,
fixed code to match BDK.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_bgx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 305cf4db14..4f830210ce 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -743,7 +743,7 @@ static void bgx_get_qlm_mode(struct bgx *bgx)
 	for (lmacid = 0; lmacid < MAX_LMAC_PER_BGX; lmacid++) {
 		int lmac_type;
 		int train_en;
-		int index = (lmacid < 2) ? 1 : 3;
+		int index = (lmacid < 2) ? 0 : 2;
 
 		lmac = &bgx->lmac[lmacid];
 
-- 
2.29.0

